src/Extensions/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,159 @@
+			this ServiceFabricHostBuilder<TContext> builder,
+			string name,
+			Func<TContext, ICommunicationListener> createListener)
+				where TContext : ServiceContext =>
+			builder.AddServiceListener(new ListenerBuilder<TContext>(name, createListener));
+
+
+		/// <summary>
+		/// Add service listener to stateless service fabric service
+		/// </summary>
+		public static ServiceFabricHostBuilder<TContext> AddServiceListener<TContext>(
+			this ServiceFabricHostBuilder<TContext> builder,
+			IListenerBuilder<TContext> listenerBuilder)
+				where TContext : ServiceContext =>
+			builder.ConfigureServices((config, collection) => collection.AddSingleton(listenerBuilder));
+
+
+		/// <summary>
+		/// Configures host to run service fabric stateless service with initializded Omex dependencies
+		/// </summary>
+		public static IHost BuildStatelessService(
+			this IHostBuilder builder,
+			string serviceName,
+			Action<ServiceFabricHostBuilder<StatelessServiceContext>> builderAction) =>
+			builder.BuildServiceFabricService<OmexStatelessServiceRunner, StatelessServiceContext>(serviceName, builderAction);
+
+
+		/// <summary>
+		/// Configures host to run service fabric stateful service with initializded Omex dependencies
+		/// </summary>
+		public static IHost BuildStatefulService(
+			this IHostBuilder builder,
+			string serviceName,
+			Action<ServiceFabricHostBuilder<StatefulServiceContext>> builderAction) =>
+			builder.BuildServiceFabricService<OmexStatefulServiceRunner, StatefulServiceContext>(serviceName, builderAction);
+
+
+		/// <summary>
+		/// Registerin Dependenci Injection classes that will provide Serfice Fabric specific information for logging
+		/// </summary>
+		public static IServiceCollection AddOmexServiceFabricDependencies<TContext>(this IServiceCollection collection)
+			where TContext : ServiceContext
+		{
+			collection.TryAddSingleton<ServiceContextAccessor<TContext>, ServiceContextAccessor<TContext>>();
+			collection.TryAddSingleton<IServiceContextAccessor<TContext>>(p => p.GetService<ServiceContextAccessor<TContext>>());
+			collection.TryAddSingleton<IServiceContextAccessor<ServiceContext>>(p => p.GetService<ServiceContextAccessor<TContext>>());
+			collection.TryAddSingleton<IServiceContext, OmexServiceFabricContext>();
+			collection.TryAddSingleton<IMachineInformation, ServiceFabricMachineInformation>();
+			return collection.AddOmexServices();
+		}
+
+
+		private static IHost BuildServiceFabricService<TRunner,TContext>(
+			this IHostBuilder builder,
+			string serviceName,
+			Action<ServiceFabricHostBuilder<TContext>> builderAction)
+				where TContext : ServiceContext
+				where TRunner : class, IOmexServiceRunner
+		{
+			try
+			{
+				builderAction(new ServiceFabricHostBuilder<TContext>(builder));
+
+				if (string.IsNullOrWhiteSpace(serviceName))
+				{
+					serviceName = Assembly.GetExecutingAssembly().GetName().FullName;
+				}
+				else
+				{
+					builder.UseApplicationName(serviceName);
```

Comment:
It a complicated one :)
For generic host `ApplicationName` is the name of the application (no connection with SF application) that it's running which in our case is a Service Fabric service so we are setting service name as an application name for the generic host.

In answer to your second question, this method is an entry point for building service, it will pass service name to SF code that will start a service, so context does not exist at this point.